frontend/app/services/api-notifications-service.js

```diff
@@ -26,37 +26,26 @@
 // See doc/COPYRIGHT.rdoc for more details.
 //++
 
-// TODO move to UI components
-module.exports = function($rootScope, $timeout, ConfigurationService) {
+module.exports = function(NotificationsService, ApiHelper) {
+  'use strict';
 
-  return {
-    restrict: 'E',
-    replace: true,
-    scope: {},
-    templateUrl: '/templates/components/flash_message.html',
-    link: function(scope, element, attrs) {
-      $rootScope.$on('flashMessage', function(event, message) {
-        scope.message = message;
-        scope.flashType = 'notice';
-        scope.flashId = 'flash-notice';
+  var addError = function(error) {
+    var messages = ApiHelper.getErrorMessages(error);
 
-        var fadeOutTime = attrs.fadeOutTime || 3000;
+    if (messages.length > 1) {
+      NotificationsService.addError('', messages);
+    }
+    else {
+      NotificationsService.addError(messages[0], []);
```

Comment:
If we move the decision of how to display the error message into the NotificationService we are in effect moving the `if` statement in there. Now that might be an improvement if we see a lot of places that call `addError` having an `if` statement before. 

I am OK with removing the necessity of having `errors`, the `message` should suffice (maybe the wording is bad as well). That would remove the awkward empty array on the `else` call.

But I think that we should otherwise leave the interface as it is. Because then we can decide on which places we want to invest into good error messages, e.g. errors on wp create/update and on which places we roll with what the API gives us, at least for the time being.

So if it is fine with you, I'd remove the necessity of having `errors`. Other than that, I wouldn't want to change it now.